__tests__/app.test.js

```diff
@@ -45,3 +43,81 @@ describe("GET /api/topics", () => {
     });
   });
 });
+
+
+
+
+describe("GET /api/articles test suite", () => {
+  test("200 status code received when calling correct api", () => {
+    return request(app).get("/api/articles").expect(200);
+  });
+
+  test("data received in the format {articles: []}", () => {
+    return request(app).get("/api/articles").expect(200)
+    .then((data) => {
+      expect(data.body).toMatchObject({
+        articles: expect.any(Object),
+      });
+      expect(Array.isArray(data.body.articles)).toBe(true);
+    });
+  });
+
+  test("there are as many articles as those in the test data", () => {
+    const numOfArticles = testData.articleData.length;
+    return request(app).get("/api/articles").expect(200)
+    .then((data) => {
+      expect(data.body.articles.length).toBe(numOfArticles);
+    });
+  });
+
+  test('each article has the correct format', () => {
+    return request(app).get("/api/articles").expect(200)
+    .then((data) => {
+      const articlesArray = data.body.articles;
+      articlesArray.forEach((article) => {
+        expect (article).toMatchObject({
+          author: expect.any(String),
+          title: expect.any(String),
+          article_id: expect.any(Number),
+          topic: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url:expect.any(String),
+          comment_count: expect.any(String),
+        });
+      });
+    });
+  });
+
+
+  test('articles are sorted by date in descending order', () => {
+    return request(app).get("/api/articles").expect(200)
+    .then((data) => {
+      const articlesArray = data.body.articles;
+      let correctOrder = true;
+      let i = 1;
+      while (correctOrder && i < articlesArray.length){
+        correctOrder = articlesArray[i-1].created_at > articlesArray[i].created_at;
+        i++;
+      }
+      expect(correctOrder).toBe(true);
+    });
```

Comment:
To make your life a bit easier, take a look at the jest sorted package, all of this can be done in one line! 🙂 https://www.npmjs.com/package/jest-sorted